kernel/kernel-api/src/test/scala/io/delta/kernel/internal/DeltaLogActionUtilsSuite.scala

```diff
@@ -287,4 +287,117 @@ class DeltaLogActionUtilsSuite extends AnyFunSuite with MockFileSystemClientUtil
+        Set(
+          FileNames.DeltaLogFileType.COMMIT,
+          FileNames.DeltaLogFileType.CHECKPOINT,
+          FileNames.DeltaLogFileType.LOG_COMPACTION).asJava,
+        dataPath,
+        startVersion,
+        endVersion,
+        false /* mustBeRecreatable */ ).toInMemoryList.asScala
+      assert(listed sameElements expectedListedFiles)
+    }
+  }
+
+  testListWithCompactions(
+    "compaction at start, no endVersion",
+    files = deltaFileStatuses(0L to 4L) ++ compactedFileStatuses(Seq((0, 4))),
+    startVersion = 0,
+    endVersion = Optional.empty(),
+    expectedListedFiles = compactedFileStatuses(Seq((0, 4))) ++ deltaFileStatuses(0L to 4L))
+
+  testListWithCompactions(
+    "compaction at end, no endVersion",
+    files = deltaFileStatuses(0L to 4L) ++ compactedFileStatuses(Seq((3, 4))),
+    startVersion = 0,
+    endVersion = Optional.empty(),
+    expectedListedFiles = deltaFileStatuses(0L to 2L) ++ compactedFileStatuses(Seq((
+      3,
+      4))) ++ deltaFileStatuses(3L to 4L))
+
+  testListWithCompactions(
+    "compaction at end, with endVersion",
+    files = deltaFileStatuses(0L to 4L) ++ compactedFileStatuses(Seq((3, 4))),
+    startVersion = 0,
+    endVersion = Optional.of(4),
+    expectedListedFiles = deltaFileStatuses(0L to 2L) ++ compactedFileStatuses(Seq((
+      3,
+      4))) ++ deltaFileStatuses(3L to 4L))
+
+  testListWithCompactions(
+    "compaction in middle, no endVersion",
+    files = deltaFileStatuses(0L to 4L) ++ compactedFileStatuses(Seq((2, 4))),
+    startVersion = 0,
+    endVersion = Optional.empty(),
+    expectedListedFiles = deltaFileStatuses(0L to 1L) ++ compactedFileStatuses(Seq((
+      2,
+      4))) ++ deltaFileStatuses(2L to 4L))
+
+  testListWithCompactions(
+    "compaction over end, with endVersion",
+    files = deltaFileStatuses(0L to 6L) ++ compactedFileStatuses(Seq((3, 7))),
+    startVersion = 0,
+    endVersion = Optional.of(5),
+    expectedListedFiles = deltaFileStatuses(0L to 5L))
+
+  testListWithCompactions(
+    "compaction before start, no endVersion",
+    files = deltaFileStatuses(0L to 6L) ++ compactedFileStatuses(Seq((2, 4))),
+    startVersion = 3,
+    endVersion = Optional.empty(),
+    expectedListedFiles = deltaFileStatuses(3L to 6L))
+
+  testListWithCompactions(
+    "compaction before start, with endVersion",
+    files = deltaFileStatuses(0L to 6L) ++ compactedFileStatuses(Seq((2, 4))),
+    startVersion = 3,
+    endVersion = Optional.of(5),
+    expectedListedFiles = deltaFileStatuses(3L to 5L))
+
+  testListWithCompactions(
+    "multiple compactions, no endVersion",
+    files = deltaFileStatuses(0L to 7L) ++ compactedFileStatuses(Seq((2, 4), (5, 7))),
+    startVersion = 0,
+    endVersion = Optional.empty(),
+    expectedListedFiles = deltaFileStatuses(0L to 1L) ++ compactedFileStatuses(Seq((
+      2,
+      4))) ++ deltaFileStatuses(2L to 4L) ++ compactedFileStatuses(Seq((
+      5,
+      7))) ++ deltaFileStatuses(5L to 7L))
```

Comment:
Darn is the scala formatter formatting it like this? with the `2, 4` on new lines?

There's no way to make this look like

```scala
 expectedListedFiles = deltaFileStatuses(0L to 1L) ++
   compactedFileStatuses(Seq((2, 4))) ++ 
   deltaFileStatuses(2L to 4L) ++
   compactedFileStatuses(Seq((5, 7))) ++ 
   ...
```

Reply:
Ohh nice. I assumed the formatted had one and only one way that it would want to format things, but yeah, I can clean it up. Thanks!